common/persistence/nosql/nosql_task_store_test.go

```diff
@@ -115,3 +115,59 @@ func TestGetTaskListSize(t *testing.T) {
 		size,
 	)
 }
+
+func TestCompleteTask(t *testing.T) {
+	store, db := setupNoSQLStoreMocks(t)
+
+	// The main assertion is that the correct parameters are passed to the DB
+	db.EXPECT().RangeDeleteTasks(gomock.Any(), &nosqlplugin.TasksFilter{
+		TaskListFilter: *getExpectedTaskListFilter(),
+		MinTaskID:      11,
+		MaxTaskID:      12,
+		BatchSize:      1,
+	}).Return(1, nil)
+
+	err := store.CompleteTask(context.Background(), &persistence.CompleteTaskRequest{
+		TaskList: &persistence.TaskListInfo{
+			DomainID: TestDomainID,
+			Name:     TestTaskListName,
+			TaskType: int(types.TaskListTypeDecision),
+		},
+		TaskID:     12,
+		DomainName: TestDomainName,
+	})
+
+	assert.NoError(t, err)
+}
+
+func TestCompleteTasksLessThan(t *testing.T) {
+	store, db := setupNoSQLStoreMocks(t)
+
+	// The main assertion is that the correct parameters are passed to the DB
+	db.EXPECT().RangeDeleteTasks(gomock.Any(), &nosqlplugin.TasksFilter{
+		TaskListFilter: *getExpectedTaskListFilter(),
```

Comment:
nit: you can just return value, then you won't need derefence.

Reply:
I need this in a later PR, so will leave it